specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2022-04-01-preview/deviceupdate.json

```diff
@@ -0,0 +1,1956 @@
+            "schema": {
+              "$ref": "#/definitions/PrivateEndpointConnectionProxyListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing the reason for operation failure.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "PrivateEndpointConnectionProxyList": {
+            "$ref": "./examples/PrivateEndpointConnectionProxies/PrivateEndpointConnectionProxy_ListByAccount.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DeviceUpdate/accounts/{accountName}/privateEndpointConnectionProxies/{privateEndpointConnectionProxyId}/validate": {
+      "post": {
+        "description": "(INTERNAL - DO NOT USE) Validates a private endpoint connection proxy object.",
+        "operationId": "PrivateEndpointConnectionProxies_Validate",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/AccountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/PrivateEndpointConnectionProxyIdParameter"
+          },
+          {
+            "name": "privateEndpointConnectionProxy",
+            "in": "body",
+            "description": "The parameters for creating a private endpoint connection proxy.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/PrivateEndpointConnectionProxy"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Validate private endpoint connection proxy empty response."
+          },
+          "default": {
+            "description": "Error response describing the reason for operation failure.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "PrivateEndpointConnectionProxyValidate": {
+            "$ref": "./examples/PrivateEndpointConnectionProxies/PrivateEndpointConnectionProxy_Validate.json"
+          }
+        },
+        "deprecated": false
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DeviceUpdate/accounts/{accountName}/privateEndpointConnectionProxies/{privateEndpointConnectionProxyId}/updatePrivateEndpointProperties": {
```

Comment:
>updatePrivateEndpointProperties [](http://example.com/codeflow?start=196&length=31)

This seems better modeled by allowing update of the remotePrivateEndpoint property in the PUT (or PATCH) payload. Any particular reason you chose to model it as a separate POST action?

Reply:
@mentat9 This is from Network RP specs. More specifically from a document that updated the spec without updating the spec document (if that makes any sense).  
[NRP spec (see section 7)](https://microsoft.sharepoint.com/:w:/r/teams/WAG/AzureNetworking/_layouts/15/Doc.aspx?sourcedoc=%7B64F10150-563A-4715-9476-FFA24F9BD42E%7D&file=%5BNRP%5D%20Private%20Endpoint%20Design%203.docx&action=default&mobileredirect=true&cid=7075b01e-18b3-42f7-950a-e1a09ede0e17)
[NRP move spec (see section 5)](https://microsoft.sharepoint.com/:w:/r/teams/Aznet/_layouts/15/doc2.aspx?sourcedoc=%7Bc9ed9692-4eb0-462d-a380-89375a2db68d%7D&action=view&wdAccPdf=0&wdparaid=36A48591&cid=0cea8de4-5888-4e69-af84-79c339fd99a9&share=IQGSlu3JsE4tRqOAiTdaLbaNASgh5QWY9yUgqtrJAkKJWjg&ovuser=72f988bf-86f1-41af-91ab-2d7cd011db47%2Cdarkoa%40microsoft.com&sourceId=&params=%7B%22AppName%22%3A%22Teams-Desktop%22%2C%22AppVersion%22%3A%2227%2F21110108719%22%7D)

I don't think any RP other than us have privateEndpointConnectionProxies API in their swagger. We have to have it because we are behind RPaaS (I can go into details, if you want).  

RPaaS does have a new feature which would allow us to remove this API from our swagger but that work is not planned yet on our side.  
RPaaS is also working on onboarding Private Endpoint Connection on behalf of all RPs.

As for the PR creation for the new version - we can do this going forward, we are planning to onboard Shift Left and, hopefully, what you are suggesting will be done automatically.